core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -601,8 +602,11 @@ private Query parse(AliasMetaData alias, QueryShardContext shardContext) {
         try {
             byte[] filterSource = alias.filter().uncompressed();
             try (XContentParser parser = XContentFactory.xContent(filterSource).createParser(filterSource)) {
-                ParsedQuery parsedFilter = shardContext.toFilter(shardContext.newParseContext(parser).parseInnerQueryBuilder());
-                return parsedFilter == null ? null : parsedFilter.query();
+                Optional<QueryBuilder> innerQueryBuilder = shardContext.newParseContext(parser).parseInnerQueryBuilder();
+                if (innerQueryBuilder.isPresent()) {
+                    return shardContext.toFilter(innerQueryBuilder.get()).query();
+                }
+                return null;
```

Comment:
That's the idea for 6.0, when we remove the possibility of having empty query clauses. Here I think we shouldn't break the current behaviour. The javadoc of the calling method (IndexService#aliasFilter) itself states that potentially returns null, which we would break if we decide to throw an error here or return MatchNone/MatchAll instead.